src/gui/render/graphics_shaders/opengl_program_status.hpp

```diff
@@ -0,0 +1,55 @@
+// -*- lsst-c++ -*-
+/*
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ */
+
+/**
+ * @file opengl_program_status.hpp
+ *
+ * @author @AlemSnyder
+ *
+ * @brief Defines program statuses
+ *
+ * @ingroup GUI  RENDER  GRAPHICS_SHADERS
+ *
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace gui {
+
+namespace shader {
+
+enum class FileStatus : uint8_t {
+    EMPTY = 0,
+    OK,
+    FILE_NOT_FOUND,
+    FILE_IO_ERROR,
+};
+
+enum class ShaderStatus : uint8_t {
+    EMPTY = 0,
+    OK,
+    INVALID_FILE,
+    COMPILATION_ERROR,
+};
+
+enum class ProgramStatus : uint8_t {
+    EMPTY = 0,
+    OK,
+    LINKING_FAILED,
+    INVALID_SHADER,
+};
```

Comment:
```suggestion
enum class FileError : uint8_t {
    OK = 0,
    EMPTY,
    FILE_NOT_FOUND,
    FILE_IO_ERROR,
};

enum class ShaderError : uint8_t {
    OK = 0,
    EMPTY,
    INVALID_FILE,
    COMPILATION_ERROR,
};

enum class ProgramError: uint8_t {
    OK = 0,
    EMPTY,
    LINKING_FAILED,
    INVALID_SHADER,
};
```

You should always have `OK = 0` so you can do `if (err)`

Reply:
> 1\. everything but 0 is an error.

That's the point.

> 2\. not sure you can cast to int

How do you think an enum is implemented? You just need to remove the `class` specifier and it does it implicitly. 

You will have to prefix the enum though: `SHADERE_OK`,

> 3\. even if you did the code would be wrong

Really. Then why does [libcurl](https://github.com/curl/curl/blob/b8c003832d730bb2f4b9de4204675ca5d9f7a903/include/curl/curl.h#L507-L509) do it? Or [zlib](https://github.com/madler/zlib/blob/f1f503da85d52e56aae11557b4d79a42bcaa2b86/zlib.h#L177-L185)? Or even the command line?